### DISCUSSION TITLE
I'm not understanding how to setup user authentication via rules

### DISCUSSION THREAD

**Discussion by dmcleish91** [pocketbase/pocketbase#984] [Q&A] [answered]
If I wanted to do a get request for a collection of **posts** for the currently signed in use and I have rule set up for the list view

``@request.user.id = userId``

how do i structure the query param or body of the post to satisfy this rule?

``http://127.0.0.1:8090/api/collections/posts/records?page=1&perPage=50``

I've been reading the docs but I'm sure something is going over my head.

**Comment by ganigeorgiev** [comment, marked as answer]
> how do i structure the query param or body of the post to satisfy this rule?

I guess you are using PocketBase v0.7.x.

Since your rule uses the `@request.user.*` filter and it requires the authenticated user to match your `userId` field, you need only to send `Authorization: User TOKEN` header alongside your GET request.

If you are using the JavaScript SDK this could be done by:

```js
import PocketBase from 'pocketbase';

const client = new PocketBase('http://127.0.0.1:8090');

await client.users.authViaEmail("test@example.com", "1234567890");

const resultList = await client.records.getList("posts", 1, 50);
```

**Reply by dmcleish91** [reply]
I was using Angular's httpClientModule and Postman to do API testing. I was not appending the auth token to my header. This instantly fixed my issue. Thank you.
